### ISSUE TITLE
xmobar hides behind windows

### ISSUE THREAD

**Issue by eazar001** [jaor/xmobar#249]
I can't see xmobar. It's always hidden behind windows, in any mode ... "tall", "wide", or "mirror tall". The only time it does not do this is when it's not in the first workspace. I find this very strange.

xmobar --version:
```
Fields missing from config defaulted: additionalFonts,textOffset,iconOffset,allDesktops,overrideRedirect,pickBroadest,iconRoot,alpha
xmobar 0.23.1
```

xmonad --version:
```
xmonad 0.12
```
If you'd like to see my xmobarrc or xmonad config file I'd be willing to post them. I'm on Arch Linux, and I've installed xmobar from the repo. I installed xmonad from Cabal. I also tried installing xmobar from Cabal ... that doesn't make a difference.

So to sum up: If I am in the first workspace ... xmobar is hidden behind application windows. If I am in any other workspace, it behaves normally. What is the reason for this? What steps can I take to resolve this?

**Comment by eazar001** [comment]
I've decided to attach screenshots describing the abovementioned behavior. Heres is the failing behavior on the first workspace:
![fail](https://cloud.githubusercontent.com/assets/5075707/12030083/b9978c1c-adad-11e5-967c-38f5df4ae64f.png)

Here is the normal/expected behavior on the second workspace:
![normal](https://cloud.githubusercontent.com/assets/5075707/12030086/c941b7e6-adad-11e5-8298-5268d469ad81.png)

**Comment by eazar001** [comment]
Hmmm, curiously I'm no longer having this issue. I'm closing this for now, this might have been some weird fluke that involved system peculiarities

**Comment by rethab** [comment]
Please reopen this ticket. I have the exact same issue.

I have checked the behavior with some older versions of xmobar, and still faced the same issue.

I am using ArchLinux with Xmonad 0.12.
The ArchLinux Xmonad package is at version 0.12-2

**Comment by eazar001** [comment]
Sure, I can reopen it. It appears the quirks have reappeared again anyway =\

**Comment by eazar001** [comment]
What I've noticed is that ... the problem dissappears if I restart `trayer` .... hence my false perception that the issue had gone away.

**Comment by skinner33** [comment]
I had the exact same problem this weekend on a new setup:
The solution, for me, was to set `overrideRedirect` to `False`, as stated in the documentation. As the default is `True`, and xmobar lists `overrideRedirect` to be missing in the configuration, I suspect you have the same issue.

**Comment by rethab** [comment]
Interesting, I have never had this set to false before, but this seems to fix the issue indeed. Thanks.

**Comment by jaor** [comment]
i think overrideRedirect was a flag introduced (very) back in the day,
specifically for xmonad, which showed this behaviour.  perhaps it was
gone for some xmonad version and it has returned... if setting
overrideRedirect as adivsed in the docs works, we can probably close
this issue, although perhaps we could make it default to true.

**Comment by jaor** [comment]
i of course meant set overrideRedirect by default to False (right now
its default is True)...

**Comment by eazar001** [comment]
I just now attempted `OverrideRedirect=False`. This works on my end as well; I'll close this issue again now.
